### PR TITLE
[7954036,7954039] Auto fix for Repo build svc account access control

### PR DESCRIPTION
**Pull request by arvindsingh-cloud**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1 [hunk 1]
```diff
@@ -230,7 +336,10 @@ class CommonSVTControls: ADOSVTBase {
                         if ($groupFoundWithExcessivePermissions -eq $true) {
                             $excessivePermissionsGroupObj = @{}
                             $excessivePermissionsGroupObj['Group'] = $broaderGroup.displayName
-                            $excessivePermissionsGroupObj['ExcessivePermissions'] = $($excessivePermissionsPerGroup.displayName -join ', ')                            
+                            $excessivePermissionsGroupObj['ExcessivePermissions'] = $($excessivePermissionsPerGroup.displayName -join '; ') 
+                            $excessivePermissionsGroupObj['Descriptor'] = $broaderGroupResponseObj[0].dataProviders.'ms.vss-admin-web.security-view-permissions-data-provider'.identityDescriptor
+                            $excessivePermissionsGroupObj['PermissionSetToken'] = $excessivePermissionsPerGroup[0].token
+                            $excessivePermissionsGroupObj['PermissionSetId'] = $excessivePermissionsPerGroup[0].namespaceId
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1:342]: We dont need to store permissionSetId, we have standard namespace in global variable $repoPermissionSetId

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1 [hunk 2]
```diff
@@ -169,6 +183,98 @@ class CommonSVTControls: ADOSVTBase {
...
+                    }
+                    $identity | Add-Member -NotePropertyName OldPermission -NotePropertyValue "Allow"
+                    $identity | Add-Member -NotePropertyName NewPermission -NotePropertyValue "Deny"
+
+                }
+            }
+            else {
+                foreach ($identity in $RawDataObjForControlFix) 
+                {
+                    $url = "https://dev.azure.com/{0}/_apis/AccessControlEntries/{1}?api-version=6.0" -f $($this.OrganizationContext.OrganizationName), $RawDataObjForControlFix[0].PermissionSetId
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1:232]: Can use permission set Id from global variable $repoPermissionSetId

File: src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1 [hunk 3]
```diff
@@ -1108,4 +1226,96 @@ class CommonSVTControls: ADOSVTBase {
...
+    hidden [ControlResult] CheckBuildSvcAcctAccessOnRepositoryAutomatedFix([ControlResult] $controlResult)
+    {
+        try{
+            $RawDataObjForControlFix = @();
+            $RawDataObjForControlFix = ([ControlHelper]::ControlFixBackup | where-object {$_.ResourceId -eq $this.ResourceId}).DataObject
+
+            if (-not $this.UndoFix) {
+                foreach ($identity in $RawDataObjForControlFix) 
+                {   
+                    $url = "https://dev.azure.com/{0}/_apis/AccessControlEntries/{1}?api-version=6.0" -f $($this.OrganizationContext.OrganizationName), $RawDataObjForControlFix[0].PermissionSetId
```
Comment (juhi037) [inline src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.CommonSVTControls.ps1:1239]: Similarily here (permission set id)

**Review by juhi037 — APPROVED** [review]

**Review by v-soukum — APPROVED** [review]
looks good

**Review by juhi037 — APPROVED** [review]

**Review by SaikumarRagala — APPROVED** [review]

**Pull request merged**